server/src/Core.hs

```diff
@@ -182,33 +168,3 @@ runSyncTimer core = mapM_ go (configSyncIntervalMicroSeconds $ coreConfig core)
     go interval = forever $ do
       enqueueCommand Sync core
       threadDelay interval
-
-persistData :: Core -> IO ()
-persistData core = do
-  (dirty, value) <- atomically $ (,) <$> readTVar (coreValueIsDirty core)
-                                     <*> readTVar (coreCurrentValue core)
-                                     <*  writeTVar (coreValueIsDirty core) False
-  when dirty $ do
-    -- persist the updated Json object to disk
-    let fileName = (configDataFile $ coreConfig core)
-        tempFileName = fileName ++ ".new"
-    -- we first write to a temporary file here and then do a rename on it
-    -- because rename is atomic on Posix and a crash during writing the
-    -- temporary file will thus not corrupt the datastore
-    writeFile tempFileName (encode value)
-    renameFile tempFileName fileName
-    forM_ (coreMetrics core) $ \m -> do
-      stat <- Posix.getFileStatus fileName
-      Metrics.setDataSize (Posix.fileSize stat) m
-      Metrics.incrementDataWritten (Posix.fileSize stat) m
```

Comment:
The invocations of `Metrics.setDataSize` and `Metrics.incrementDataWritten` seem to have been removed.

Reply:
We can split the data written metric into data file written and journal written.

Reply:
Oh, good catch! I'll fix that.

Reply:
Metrics are added back, separated by journal bytes written and data bytes written.